libheif/heif_emscripten.h

```diff
@@ -259,6 +259,95 @@ static emscripten::val heif_js_decode_image2(struct heif_image_handle* handle,
   return result;
 }
 
+static emscripten::val heif_js_depth_img_decode(struct heif_image_handle *handle, heif_item_id depth_image_id)
+{
+  emscripten::val depth_result = emscripten::val::object();
+  struct heif_image_handle *depth_handle;
+  heif_image_handle_get_depth_image_handle(handle, depth_image_id, &depth_handle);
+  if (!depth_handle)
+  {
+    depth_result.set("err", "could not get handle of depth image");
+    depth_result.set("id", depth_image_id);
+    return depth_result;
+  }
```

Comment:
It's better to return an `heif_error` here since heif_error is registered as an value object [see here](https://github.com/strukturag/libheif/blob/a3f43c1b67fbdd9892cc101d231029d4fde10c19/libheif/api/libheif/heif_emscripten.h#L463C1-L463C55), that way it won't be necessary to manually delete the object on the JS side [see here](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html?highlight=embind#memory-management)